flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReader.java

```diff
@@ -224,6 +236,20 @@ public void addSplits(List<MySqlSplit> splits) {
                 }
             } else {
                 MySqlBinlogSplit binlogSplit = split.asBinlogSplit();
+                // When restore from a checkpoint, the finished split infos may contain some splits
+                // for the deleted tables.
+                // We need to remove these splits for the deleted tables at the finished split
+                // infos.
+                if (!checkTableChangeForBinlogSplit) {
```

Comment:
wait, here should be `if(checkTableChangeForBinlogSplit)`, we should update the code where this method is called

Reply:
Yes, I have fixed this part in the newest commit.